src/preferred-provider.js

```diff
@@ -94,16 +122,30 @@ function discoverFromHeaders (webId) {
     })
 }
 
-function discoverFromProfile (webId) {
+function discoverAllFromProfile (webId) {
```

Comment:
I think this should then also become `validateFromProfile` and accept a second `issuer` parameter. Will eliminate most of the messy logic above.